From fcee36d38d21276f2cb172b89e9f5258ecf8278b Mon Sep 17 00:00:00 2001
From: Thor Thayer <tthayer@opensource.altera.com>
Date: Wed, 15 Jul 2015 17:01:57 -0500
Subject: [PATCH 149/172] FogBugz #311020-4: update altera_edac.h

Corrections from merge of A10 DDR support with upstream and downstream
kernels during v4.1 upgrade. Both trees contained altera_edac.h file. The
merge of the A10 DDR support overwrote the L2 EDAC and OCRAM EDAC defines.
Put them back in this commit.

Signed-off-by: Thor Thayer <tthayer@opensource.altera.com>
[czou:Original patch taken from 
https://github.com/altera-opensource/linux-socfpga.git socfpga-4.1]
Signed-off-by: czou <cao.zou@windriver.com>
---
 drivers/edac/altera_edac.h | 31 +++++++++++++++++++++++++++++++
 1 file changed, 31 insertions(+)

diff --git a/drivers/edac/altera_edac.h b/drivers/edac/altera_edac.h
index 7b64dc7..a6e334f 100644
--- a/drivers/edac/altera_edac.h
+++ b/drivers/edac/altera_edac.h
@@ -19,7 +19,9 @@
 #define _ALTERA_EDAC_H
 
 #include <linux/edac.h>
+#include <linux/platform_device.h>
 #include <linux/types.h>
+#include "edac_core.h"
 
 /* SDRAM Controller CtrlCfg Register */
 #define CV_CTLCFG_OFST             0x00
@@ -198,4 +200,33 @@ struct altr_sdram_mc_data {
 	const struct altr_sdram_prv_data *data;
 };
 
+struct ecc_mgr_prv_data {
+	int (*setup)(struct platform_device *pdev, void __iomem *base);
+	int ce_clear_mask;
+	int ue_clear_mask;
+#ifdef CONFIG_EDAC_DEBUG
+	struct edac_dev_sysfs_attribute *eccmgr_sysfs_attr;
+	void * (*init_mem)(size_t size, void **other);
+	void (*free_mem)(void *p, size_t size, void *other);
+	int ecc_enable_mask;
+	int ce_set_mask;
+	int ue_set_mask;
+	int trig_alloc_sz;
+#endif
+};
+
+struct altr_ecc_mgr_dev {
+	void __iomem *base;
+	int sb_irq;
+	int db_irq;
+	const struct ecc_mgr_prv_data *data;
+	char *edac_dev_name;
+};
+
+extern const struct ecc_mgr_prv_data l2ecc_data;
+extern const struct ecc_mgr_prv_data ocramecc_data;
+
+ssize_t altr_ecc_mgr_trig(struct edac_device_ctl_info *edac_dci,
+			  const char *buffer, size_t count);
+
 #endif	/* #ifndef _ALTERA_EDAC_H */
-- 
1.9.1

